File: src/main/java/no/nav/veilarbaktivitet/avtalt_med_nav/Forhaandsorientering.java
package no.nav.veilarbaktivitet.avtalt_med_nav;

import lombok.Builder;
import lombok.Data;
import no.nav.common.types.identer.AktorId;

import java.util.Date;

@Data
@Builder(toBuilder = true)
public class Forhaandsorientering {

    private String id;
    private Type type;
    private String tekst;
    private Date lestDato;
    private AktorId aktorId;
    private String arenaAktivitetId;
    private String aktivitetId;
    private String aktivitetVersjon;
    private Date opprettetDato;
    private String opprettetAv;
    private String varselId;
    private Date varselSkalStoppesDato;
    private Date varselStoppetDato;

    public static ForhaandsorienteringBuilder builder() {
        return new CustomForhaandsorienteringBuilder();
    }

    public ForhaandsorienteringDTO toDTO() {
        return new ForhaandsorienteringDTO(id, type, tekst, lestDato).toBuilder().build();
    }

    public static class CustomForhaandsorienteringBuilder extends ForhaandsorienteringBuilder {
        @Override
        public Forhaandsorientering build() {
            if (super.type == null) {
                return null;
            }

            return super.build();
        }
    }
}

File: src/main/kotlin/no/nav/veilarbaktivitet/avtalt_med_nav/Forhaandsorientering.kt
package no.nav.veilarbaktivitet.avtalt_med_nav

import no.nav.veilarbaktivitet.person.Person
import java.util.*

data class Forhaandsorientering(
    val id: String,
    val type: Type,
    val tekst: String,
    val lestDato: Date?,
    val aktorId: Person.AktorId,
    val arenaAktivitetId: String?,
    val aktivitetId: String,
    val aktivitetVersjon: String,
    val opprettetDato: Date,
    val opprettetAv: String,
    val varselId: String,
    val varselSkalStoppesDato: Date?,
    val varselStoppetDato: Date?,
) {
    fun toDTO() = ForhaandsorienteringDTO(id, type, tekst, lestDato)
}
